File: Python/162__Find_Peak_Element.py
class Solution(object):
    def findPeakElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0:
            return -1
        if len(nums) == 1:
            return 0
        for i in range(len(nums)):
            if i == 0:
                if nums[0] > nums[1]:
                    return 0
            if i == len(nums) - 1:
                if nums[-1] >= nums[-2]:
                    return len(nums) - 1
            if nums[i] > nums[i - 1] and nums[i] > nums[i + 1]:
                return i
        

File: Java/162__Find_Peak_Element.java
class Solution {
    public int findPeakElement(int[] nums) {
        if (nums.length == 1) {
            return 0;
        }
        if (nums.length == 2) {
            if (nums[0] == nums[1]) {
                return -1;
            } else if (nums[0] > nums[1]) {
                return 0;
            } else {
                return 1;
            }
        }
        if (nums[0] > nums[1]) {
            return 0;
        }
        if (nums[nums.length - 1] > nums[nums.length - 2]) {
            return nums.length - 1;
        }
        int i = 0;
        while (i < nums.length) {
            while (i < nums.length - 1 && nums[i] <= nums[i + 1]) {
                i++;
            }
            if (nums[i] != nums[i - 1]) {
                return i;
            }
            while (i < nums.length - 1 && nums[i] >= nums[i + 1]) {
                i++;
            }
        }
        return -1;

    }
}
